config/samples/bgp_full_sample_config.yaml

```diff
@@ -0,0 +1,54 @@
+
+---
+apiVersion: metallb.io/v1alpha1
+kind: AddressPool
+metadata:
+  name: addresspool-sample1
+  namespace: metallb-system
+spec:
+  protocol: bgp
+  addresses:
+    - 172.18.0.100-172.18.0.255
+  bgpAdvertisements:
```

Comment:
Have you looked into a validation webhook yet?  I wouldn't block on it, but this is a case where I think we'd want to reject a config that sets config in this section without `protocol: bgp`.

Reply:
not yet webhook been deferred to 4.10 its on my todo list once we flush all configs we need to compile list of validation needed then go after them, I will take a not of this in the open issue for webhook

Reply:
sounds good